.github/workflows/native_toolchain_c.yaml

```diff
@@ -0,0 +1,59 @@
+name: dart
+permissions: read-all
+
+on:
+  pull_request:
+    branches: [main]
+    paths:
+      - ".github/workflows/native_toolchain_c.yaml"
+      - "pkgs/native_toolchain_c/**"
+  push:
+    branches: [main]
+    paths:
+      - ".github/workflows/native_toolchain_c.yaml"
+      - "pkgs/native_toolchain_c/**"
+  schedule:
+    - cron: "0 0 * * 0" # weekly
+
+jobs:
+  clang-bleeding-edge:
+    strategy:
+      matrix:
+        os: [ubuntu]
+        sdk: [stable]
+        package: [native_toolchain_c]
+
+    runs-on: ${{ matrix.os }}-latest
+
+    defaults:
+      run:
+        working-directory: pkgs/${{ matrix.package }}
+
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744
+
+      - uses: dart-lang/setup-dart@d6a63dab3335f427404425de0fbfed4686d93c4f
+        with:
+          sdk: ${{ matrix.sdk }}
+
+      - uses: nttld/setup-ndk@dbacc5871a0fac6eef9a09d2ca86bc8bf79432c3
+        with:
+          ndk-version: r25b
+        if: ${{ matrix.sdk == 'stable' }}
+
+      - run: dart pub get
+
+      - name: Install native toolchains
+        run: sudo apt-get update && sudo apt-get install gcc-i686-linux-gnu gcc-aarch64-linux-gnu gcc-arm-linux-gnueabihf
+        if: ${{ matrix.sdk == 'stable' && matrix.os == 'ubuntu' }}
+
+      - run: git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git
+      - run: echo "$PWD/depot_tools" >> $GITHUB_PATH
+      - run: mkdir dart-sdk
+      - run: cd dart-sdk && fetch --no-history dart
+      - run: ./dart-sdk/sdk/buildtools/linux-x64/clang/bin/clang --version
```

Comment:
I just added that to show it materializes the SDK clang, seems redundant given the next two lines.